lib/tapioca/runtime/trackers/required_ancestor.rb

```diff
@@ -41,6 +41,7 @@ module T
   module Helpers
     prepend(Module.new do
       def requires_ancestor(&block)
+        block = T.cast(block, T.proc.returns(Module))
```

Comment:
Why is this needed and why can't we just use `T.unsafe`? After all, this is a patch that has no attempt at being typed and doesn't benefit much from it.

Reply:
Yeah, that's the general approach but, in this case, it doesn't really get us any safety and is just a runtime overhead.

I am still curious as to why this suddenly needs to be typed, though.

Reply:
I think that's affected by [this change](https://github.com/sorbet/sorbet/pull/6299/files#r955406567)? cc @egiurleo

Reply:
Ah ok, In that case what we should be doing is widen the signature of the block that the receiving method is expecting.

This patch exists to store the blocks from requires_ancestor calls, to later generate those calls in RBI files properly.

So, at this point we need to make sure that we can generate these new kinds of requires_ancestor calls properly too. So this PR need to extend the current test that is generating those with something like `requires_ancestor { T.class_of(String) }` and assert that we generate the same in the output. That might mean changing some expected signatures too.

Reply:
I added `requires_ancestor { T.class_of(String) }` to the tests, but I think it's generated correctly before/after the Sorbet upgrade.